### PR TITLE
Menus persistence layer - data models and database methods for Menus

### PR DESCRIPTION
**Pull request by tonyr59h**
First step of the Menus process. #3588

As with Site Settings, all Menus work will be merged into a feature branch until it's ready to ship.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/DatabaseUtils.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
+package org.wordpress.android.util;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+
+import java.util.List;
+
+public class DatabaseUtils {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/util/DatabaseUtils.java]: We should merge this with `SqlUtils` (or move it to the wputils subtree).

**Comment by maxme** [comment]
@tonyr59h just one comment here, and it's not super important - I wonder how you're going to implement the UI? Using a `RecyclerView` + `ItemTouchHelper`?

File: WordPress/src/main/java/org/wordpress/android/util/DatabaseUtils.java (hunk 1 above)
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/util/DatabaseUtils.java]: Good point, I'll move them to `SqlUtils`.

**Comment by tonyr59h** [comment]
@maxme `ItemTouchHelper` was my running thought but I haven't looked into alternatives yet. I'll take a look around the web when I finish up the networking layer. Do you know of any good libraries that we might be able to use?

**Comment by maxme** [comment]
> Do you know of any good libraries that we might be able to use?

No. `ItemTouchHelper` seems great and it lets room for customization, but I don't know if that's enough for the Menu implementation.

**Comment by tonyr59h** [comment]
Marked as `Not Ready for Review` while I double check the models against the iOS project implementation.

**Comment by tonyr59h** [comment]
Gonna close this and open two PR's to separate the tests and feature work.

**Pull request closed without merging**